librad/src/id.rs

```diff
@@ -98,18 +138,21 @@ where
     fn resolve(&self, uri: &RadicleUri) -> Result<E, Error>;
 }
 
-pub trait RevisionsResolver<E>
+pub trait RevisionsResolver<E, I, II>
 where
     E: Entity,
+    I: Iterator<Item = E>,
+    II: IntoIterator<Item = E, IntoIter = I> + Sized,
 {
-    fn resolve_revisions(&self, uri: &RadicleUri) -> Box<dyn Iterator<Item = &E>>;
+    fn resolve_revisions(&self, uri: &RadicleUri) -> Box<II>;
 }
 
-pub trait Entity {
+pub trait Entity: Sized {
```

Comment:
I find this trait a bit confusing with all the functions and a lot of them having defaults that don't use `self`. For example `check_history<E, I, II>` can live outside of the `trait`.

Would it be possible to trim down the functions to a necessary set? Maybe this is just a stylistic thing since I just get that "fishy" feeling :sweat_smile: